packages/editor-ui/src/stores/templates.store.ts

```diff
@@ -63,7 +63,7 @@ export const useTemplatesStore = defineStore(STORES.TEMPLATES, {
 			return (id: string): null | ITemplatesCollection => this.collections[id];
 		},
 		getCategoryById() {
-			return (id: string): null | ITemplatesCategory => this.categories[id];
+			return (id: string): null | ITemplatesCategory => this.categories[id as unknown as number];
```

Comment:
Instead of coercing the type here we should change it to expect id: number and update the only place where this is called to not pass it as string(`packages/editor-ui/src/views/TemplatesSearchView.vue#283`): 
```ts
categories_applied: categories.map((categoryId) =>
					this.templatesStore.getCategoryById(categoryId),
				) as ITemplatesCategory[],
```

Reply:
The whole `this.categories` is a mess in `templates.store.ts`. Sometimes it's an object, sometimes it's an array. We should fix this separately.